voyager.js

```diff
@@ -45,31 +45,64 @@ new BasicTracerProvider().register()
 Zinnia.activity.info('Voyager benchmarking started')
 
 export async function runSaturnBenchmarkInterval() {
-    const random = Math.random()
-    if (random <= prodOpts.sampleRate) {
-        console.log('Running prod benchmark...')
-        try {
-            await runBenchmark(prodSaturn)
-            console.log('Prod benchmark successful!')
-        } catch (err) {
-            console.error(err)
-        }
-        Zinnia.jobCompleted()
+    let runningOnSaturnNode = false
+    try {
+        runningOnSaturnNode = await isRunningOnSaturnNode()
+    } catch (err) {
+        console.error(
+            'Could not check if running on Saturn node',
+            { cause: err }
+        )
     }
-    if (random <= testOpts.sampleRate) {
-        console.log('Running test benchmark...')
-        try {
-            await runBenchmark(testSaturn)
-            console.log('Test benchmark successful!')
-        } catch (err) {
-            console.error(err)
+    if (runningOnSaturnNode) {
+        console.log('Running on Saturn host, skipping benchmark')
+    } else {
+        const random = Math.random()
+        if (random <= prodOpts.sampleRate) {
+            console.log('Running prod benchmark...')
+            try {
+                await runBenchmark(prodSaturn)
+                console.log('Prod benchmark successful!')
+            } catch (err) {
+                console.error(err)
+            }
+            Zinnia.jobCompleted()
+        }
+        if (random <= testOpts.sampleRate) {
+            console.log('Running test benchmark...')
+            try {
+                await runBenchmark(testSaturn)
+                console.log('Test benchmark successful!')
+            } catch (err) {
+                console.error(err)
+            }
+            Zinnia.jobCompleted()
         }
-        Zinnia.jobCompleted()
     }
     console.log('Sleeping for 60s...')
     setTimeout(runSaturnBenchmarkInterval, 1000 * 60)
 }
 
+async function getPublicIPv4Address () {
+    // TODO Replace with Voyager API once available
+    const res = await fetch(`https://api.filspark.com/inspect-request`)
+    const { cloudfareAddr: ip } = await res.json()
+    return ip
+}
+
+async function isRunningOnSaturnNode () {
+    const ip = await getPublicIPv4Address()
+    const subdomain = ip.replaceAll('.', '-')
+    try {
+        await fetch(`https://${subdomain}.l1s.saturn.ms/`, {
```

Comment:
I'm pretty sure this is enough, at least for now, but I want @hannahhoward to double check my reasoning that if this returns anything at all there is a Saturn node running on the given IP - couldn't think of any cases where that wouldn't be true (aside from maybe some node startup/shutdown edge cases that probably aren't worth worrying about right now?) but also worth noting that I'm still feeling my way around the Saturn codebase so might be missing something!

This definitely seems worth shipping now in any case - we can refine as we go!

Reply:
Ok, let's merge if there is pressure to do so, otherwise wait for @hannahhoward's review?

Reply:
I think it's totally fine to wait, this isn't "need it today" urgent as far as I know!

Reply:
have we tested this against an actual saturn node to verify it accepts an HTTP connection at root?

Reply:
If this doesn't end up working, you can definitely use `/ipfs` with the health check cid -- https://github.com/filecoin-saturn/orchestrator/blob/651e34809ac02cf174f6fa9444524b8624132b1c/src/cron/health-check.js#L26

Reply:
I've tested this with an actual Saturn node, and it worked 👍

Reply:
On the root address, it will redirect to saturn.ms, so it's a good http response code